framework/wazuh/agent.py

```diff
@@ -703,14 +703,36 @@ def assign_agents_to_group(group_list=None, agent_list=None, replace=False, repl
     # Check if the group exists
     if not Agent.group_exists(group_id):
         raise WazuhResourceNotFound(1710)
-    system_agents = get_agents_info()
+
+    with WazuhDBQueryAgents(limit=None, select=['id', 'status']) as db_query:
```

Comment:
This query does not seem to scale very well in highly loaded environments. We query all agents always.